packages/core/test/unit/query-generator/select-query.test.ts

```diff
@@ -675,4 +676,102 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- we are testing invalid values
+        indexHints: [{ type: 'INVALID', values: ['index_project_on_name'] }],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['indexHints']),
+        'mariadb mysql snowflake': 'SELECT [id] FROM [Users] AS [User];',
+      });
+    });
+  });
+
+  describe('table hints', () => {
+    it('support an array of table hints', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        tableHints: [TableHints.UPDLOCK, TableHints.PAGLOCK],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['tableHints']),
+        mssql: `SELECT [id] FROM [Users] AS [User] WITH (UPDLOCK, PAGLOCK);`,
+      });
+    });
+
+    it('should be able to use table hints on joins', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        tableHints: [TableHints.NOLOCK],
+        include: _validateIncludedElements({
+          model: User,
+          include: [{
+            association: User.associations.projects,
+            attributes: ['id'],
+          }],
+        }).include,
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['tableHints']),
+        mssql: `SELECT [User].[id], [projects].[id] AS [projects.id] FROM [Users] AS [User] WITH (NOLOCK) LEFT OUTER JOIN [Projects] AS [projects] WITH (NOLOCK) ON [User].[id] = [projects].[UserId];`,
+      });
+    });
+
+    it('should be able to use separate table hints on joins', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        tableHints: [TableHints.NOLOCK],
+        include: _validateIncludedElements({
+          model: User,
+          include: [{
+            association: User.associations.projects,
+            attributes: ['id'],
+            tableHints: [TableHints.READPAST],
+          }],
+        }).include,
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['tableHints']),
+        mssql: `SELECT [User].[id], [projects].[id] AS [projects.id] FROM [Users] AS [User] WITH (NOLOCK) LEFT OUTER JOIN [Projects] AS [projects] WITH (READPAST) ON [User].[id] = [projects].[UserId];`,
+      });
+    });
+
+    it('should not add a table hint if the type is not valid', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- we are testing invalid values
+        tableHints: ['INVALID'],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['tableHints']),
+        mssql: 'SELECT [id] FROM [Users] AS [User];',
```

Comment:
Same here, maybe we should throw on invalid tableHints?

Reply:
Good idea, updated now